foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/QueryExecutor.java

```diff
@@ -45,14 +45,23 @@ public QueryExecutor(AnalyticsLoader analyticsLoader, ExecutorService executorSe
     }
 
     public <T extends ActionRequest> ActionResponse execute(T request) throws FoxtrotException {
+        validate(request);
         Action action = resolve(request);
         return action.execute();
     }
 
     public <T extends ActionRequest> AsyncDataToken executeAsync(T request) throws FoxtrotException {
+        request.validate();
         return resolve(request).execute(executorService);
     }
 
+    public <T extends ActionRequest> void validate(T request) throws FoxtrotException {
```

Comment:
Hmmm .. this is already there, create an abstract validateImpl(request) method, so that it can be consolidated in the action, rather than the model class.